NiBAx/plugins/harmonizationplugin/harmonizationplugin.py

```diff
@@ -0,0 +1,378 @@
+        for count,site in enumerate(parameters['SITE']):
+            self.plotCanvas.axes2.get_yticks()
+            if 'nan' in parameters.loc[site]['gamma_values']:
+                b.text(-0.05,self.plotCanvas.axes2.get_yticks()[count]-0.2,' nan',fontsize='x-small',ha='right')
+            else:
+                b.text(-0.05,self.plotCanvas.axes2.get_yticks()[count]-0.2,parameters.loc[site]['gamma_values'],fontsize='x-small',ha='right')
+                b.text(0.05,self.plotCanvas.axes2.get_yticks()[count]-0.2,parameters.loc[site]['delta_values'],fontsize='x-small',ha='left')
+        b.set_xlabel('')
+        b.set_ylabel('')
+        b.set(yticklabels=[])
+        self.plotCanvas.axes2.xaxis.set_ticks_position('bottom')
+        b.tick_params(axis='both',left=False,right=False, length=4)
+        self.plotCanvas.axes2.axvline(0,color='black',linewidth=0.25)
+        sns.despine(ax=self.plotCanvas.axes2, left=True)
+
+        # get title as ROI name relative to scale/shift label
+        _, MUSEDictIDtoNAME = self.datamodel.GetMUSEDictionaries()
+        title = currentROI
+        if title.startswith('MUSE_'):
+            title = '(MUSE) ' + list(map(MUSEDictIDtoNAME.get, [currentROI]))[0]
+
+        if title.startswith('WMLS_'):
+            title = '(WMLS) ' + list(map(MUSEDictIDtoNAME.get, [currentROI.replace('WMLS_', 'MUSE_')]))[0]
+
+        self.plotCanvas.axes2.get_figure().subplots_adjust(top=.8)
+        self.plotCanvas.axes2.get_figure().suptitle(title)
+        
+        
+        self.plotCanvas.axes3.get_figure().set_tight_layout(True)
+        self.plotCanvas.axes3.set_xlim(-4*sd_raw, 4*sd_raw)
+        self.plotCanvas.axes3.set_ylim( self.plotCanvas.axes1.get_ylim() )
+        sns.set(style='white')
+        c = sns.boxplot(x=h_res, y="SITE", data=data, palette=cSite,linewidth=0.25,showfliers = False,ax=self.plotCanvas.axes3,**PROPS)
+        nobs2 = data['SITE'].value_counts().sort_index(ascending=True).values
+        nobs2 = [str(x) for x in nobs2.tolist()]
+        nobs2 = [i for i in nobs2]
+        if nobs1 != nobs2:
+            print('not equal sample sizes')
+        a.set_yticklabels(labels)
+        self.plotCanvas.axes3.axvline(ci_plus_h,color='grey',ls='--')
+        self.plotCanvas.axes3.axvline(ci_minus_h,color='grey',ls='--')
+        self.plotCanvas.axes3.yaxis.set_ticks_position('left')
+        self.plotCanvas.axes3.xaxis.set_ticks_position('bottom')
+        c.tick_params(axis='both', which='major', length=4)
+        c.set_xlabel('Residuals after harmonization')
+        c.set_ylabel('')
+        c.set(yticklabels=[])
+        sns.despine(ax=self.plotCanvas.axes1, trim=True)
+        sns.despine(ax=self.plotCanvas.axes3, trim=True)
+
+        self.plotCanvas.draw()
+
+    def OnAddHarmonizedMUSE(self):
+        Harmonize(self.datamodel, self.datamodel.harmonization_model).AddHarmonizedMUSE(self.MUSE)
```

Comment:
There should be only one instance of Harmonize class, another one is on line 145. Please use composition.